### DISCUSSION TITLE
qBitrr - Connections Quests

### DISCUSSION THREAD

**Discussion by LegioN7** [Feramance/qBitrr#98] [Q&A] [answered]
Hello Feramance,
I wanted to ask how qBitrr handles various connections.

If a file is being downloaded by qBittorrent, and it has 0 seeds (assuming this is going to just stall out)
Does qBitrr recognize this?

If a torrent is active for  xyz timer, does the freespace setting override this timer? I have some files active that are not being removed, and in the qBitrr window, I can see it mentioning the space requirements, but not the connection/speed/time requirements.

Thank you

**Comment by Feramance** [comment, marked as answer]
Hi @LegioN7,

Not sure if it's what you meant but, qBitrr waits until the torrent older than the `ignore torrents younger than ...` config in the config file, then checks if the torrent is stalled or not, even with 0 seeds. Both stalled and metadata download (when it takes too long) are treated as stalled.

With regards to the free space logic, it takes precedence over everything, if there isn't enough space to download then pause it and don't process it.

Hope these answer your questions

**Comment by LegioN7** [comment]
Hey Feramance,
I wanted to write back but have been busy! In this case, the reason I ask would be, wouldn't these type of files be "dead" by nature?
It would be awesome to identify these 0 seeder torrents and consider them stalled/inactive.

![image](https://github.com/user-attachments/assets/51438019-3a17-49c5-a3b3-9bb09006a695)

**Comment by Feramance** [comment]
By default it gives them 10 mins to attempt to connect to seeders, if not then it removes and blocks them. This is already a fact

**Comment by Feramance** [comment]
I did notice a small bug which I have now fixed, update to v4.9.0 and let me know how it goes. Thanks
